src/components/Questions/Palette/QuestionPalette.vue

```diff
@@ -1,46 +1,71 @@
 <template>
   <div class="bg-white p-4 sm:p-6 lg:p-8 overflow-auto sm:w-1/3 lg:w-1/3 xl:w-1/3">
-    <div
-      class="bg-gray-200 rounded-md p-4 grid grid-rows-2 space-y-2"
-    >
-      <div class="grid grid-cols-2">
-        <Success
-          :title="legendSuccessText"
-          :hasQuizEnded="hasQuizEnded"
-        ></Success>
-        <Error :title="legendErrorText" :hasQuizEnded="hasQuizEnded"></Error>
-      </div>
-
-      <div class="grid grid-cols-2">
-        <Neutral
-        :title="legendNeutralText"
-        :hasQuizEnded="hasQuizEnded"
-      ></Neutral>
-        <div v-if="hasQuizEnded">
-          <PartialSuccess :title="legendPartialSuccessText" :hasQuizEnded="hasQuizEnded"></PartialSuccess>
-        </div>
-      </div>
+    <div class="inline-flex rounded-md w-full" role="group">
+      <button
+        :class="togglePaletteButtonClass"
+        class="mr-1"
+        type="button"
+        @click="togglePalette"
+      >PALETTE</button>
+      <button
+        :class="toggleInstructionsButtonClass"
+        class="ml-1"
+        type="button"
+        @click="toggleInstructions"
+      >INSTRUCTIONS</button>
     </div>
-
-    <div
-      v-for="(questionSetState, index) in questionSetStates" :key="index" class="space-y-2">
-        <p :class="titleTextClass" :data-test="`paletteTitle-${index}`">{{ questionSetState.title }}</p>
-        <div :class="instructionTextClass" :data-test="`paletteInstruction-${index}`" v-html="questionSetState.instructionText"></div>
-        <div class="grid grid-cols-5 bp-500:grid-cols-6 lg:grid-cols-7 xl:grid-cols-8 mt-4 space-y-4">
-          <PaletteItem
-            v-for="(questionState, qindex) in questionSetState.paletteItems"
-            class="hover:cursor-pointer"
-            :class="{ 'mt-4': qindex == 0 }"
-            :key="qindex"
-            :index="questionState.index"
+      <InstructionPage
+      v-if="showInstructionButton"
+      :title="title"
+      :subject="subject"
+      :testFormat="testFormat"
+      :maxMarks="maxMarks"
+      :max-questions-allowed-to-attempt="numQuestions"
+      :quiz-time-limit="quizTimeLimit"
+      :questionSets = "questionSets"
```

Comment:
i see that InstructionPage uses `questionSets` . Is it possible to make do with `questionSetStates` ? See `Player.vue` for `questionSetStates` computation. My sense is that `questionSetStates` would likely be sufficient

Reply:
there are two flows to instructionpage

player -> splash -> instructionpage

player -> questionmodal -> body -> palette -> instructionpage

my sense is, `questionSetStates` would be sufficient to pass, which is being computed in player.vue

Reply:
Replaced questionSets to questionSetStates from the instructionPage component.
Yes questionSetStates were sufficient.